compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -569,6 +569,17 @@ void addMLIRAIRAIELoweringPasses(OpPassManager &passManager, bool packPeel) {
   passManager.addPass(createCanonicalizerPass());
 }
 
+// NOTE: this runs on the top-level program module containing all hal.executable
+// ops.
+void buildAMDAIELinkingPassPipeline(OpPassManager &passManager) {
+  // Link together executables. This may produce some IR duplication.
+  passManager.addPass(createAMDAIELinkExecutablesPass());
+
+  // Cleanup IR duplication.
+  passManager.addNestedPass<IREE::HAL::ExecutableOp>(
+      mlir::createCanonicalizerPass());
+}
```

Comment:
Why do you expose this in Passes.cpp? Can it be directly added to [AIETarget.cpp](https://github.com/nod-ai/iree-amd-aie/pull/395/files#diff-2a27c2baf66b75bcf426d927b55cbf5f3b5d7bfa69b9b7492ab0d3672805bac4R151)?

Reply:
Good point, this is just the pattern I saw accross all the other backends, I think the *Target.cpp files just call the top level pass pipelines and they are implemented in the Passes.cpp file. e.g
https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/SPIRV/Passes.cpp#L663-L681
https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp#L776-L789